main.go

```diff
@@ -52,80 +52,123 @@ var (
 	reportFile       = "report.txt"
 	dbNameOverride   string
 	instanceOverride string
+	projectOverride  string
 	filePrefix       = ""
 	verbose          bool
 )
 
+func init() {
+	flag.StringVar(&dbNameOverride, "dbname", "", "dbname: name to use for Spanner db")
+	flag.StringVar(&instanceOverride, "instance", "", "instance: Spanner instance to use")
+	flag.StringVar(&projectOverride, "project", "", "project: Google Cloud project to use")
+	flag.StringVar(&filePrefix, "prefix", "", "prefix: file prefix for generated files")
+	flag.BoolVar(&verbose, "v", false, "verbose: print additional output")
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+	flag.PrintDefaults()
+	fmt.Fprintf(os.Stderr, `Note: input is always read from stdin.
+Sample usage:
+  pg_dump mydb | %s
+  %s < my_pg_dump_file
+`, os.Args[0], os.Args[0])
+}
+
 func main() {
 	flag.Usage = usage
 	flag.Parse()
 	internal.VerboseInit(verbose)
-	now := time.Now()
 	lf, err := setupLogFile()
 	if err != nil {
 		fmt.Printf("\nCan't set up log file: %v\n", err)
 		panic(fmt.Errorf("can't set up log file"))
 	}
 	defer close(lf)
-	project, err := getProject()
+
+	err = process(projectOverride, instanceOverride, dbNameOverride, nil, filePrefix)
 	if err != nil {
-		fmt.Printf("\nCan't get project: %v\n", err)
-		panic(fmt.Errorf("can't get project"))
+		panic(fmt.Errorf("failed to run the job: %v", err))
```

Comment:
Not sure "failed to run the job" is the right message here e.g. not clear what "job" means in this context.
Maybe we just use the err returned and don't wrap it in anything else?